### DISCUSSION TITLE
Tuya smart plug power usage

### DISCUSSION THREAD

**Discussion by WoutervanderLoopNL** [arendst/Tasmota#13962] [Ideas]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
I there, I've bought a [Tuya Smart Zigbee Plug](https://nl.aliexpress.com/item/1005003490077598.html?spm=a2g0s.9042311.0.0.7d1f4c4dWD1qFM)  Which should be compatible with Zigbee. 

ive entered the device in pairing mode, and I permitted the join in the Zigbee Bridge. 
The device is added to the bridge: 
![image](https://user-images.githubusercontent.com/8301044/145244786-dba00dc5-cb30-4a8a-8477-a69f86ff8c54.png)
 and it receive status updates when the device is on or off. 
 
 but i don't get any updates about the current power useage of the device.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ ] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): SONOFF ZBBridge - Smart Zigbee Bridge
- [x] Tasmota binary firmware version number used: Tasmota 10.0.0.4 by Theo Arends
  - [x] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [x] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
17:25:45.885 MQT: stat/tasmota/RESULT = {"NAME":"Generic","GPIO":[1,1,1,1,1,1,1,1,1,1,1,1,1,1],"FLAG":0,"BASE":18}
17:25:46.115 MQT: stat/tasmota/RESULT = {"Module":{"75":"Sonoff ZbBridge"}}
17:25:46.369 MQT: stat/tasmota/RESULT = {"GPIO0":{"320":"Led_i1"},"GPIO1":{"3552":"Zigbee Tx"},"GPIO2":{"0":"None"},"GPIO3":{"3584":"Zigbee Rx"},"GPIO4":{"5312":"Zigbee Rst1"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"640":"I2C SDA"},"GPIO13":{"576":"LedLink_i"},"GPIO14":{"608":"I2C SCL"},"GPIO15":{"0":"None"},"GPIO16":{"32":"Button1"},"GPIO17":{"0":"None"}}


```
- [x] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:
17:26:21.355 MQT: stat/tasmota/RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":114,"Free":397,"Rules":"on Switch1#State=1 do Publish stat/%topic%/DOOR ON endon on Switch1#State=0 do Publish stat/%topic%/DOOR OFF endon"}}
17:26:21.588 MQT: stat/tasmota/RESULT = {"Rule2":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}
17:26:21.838 MQT: stat/tasmota/RESULT = {"Rule3":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":0,"Free":511,"Rules":""}}

```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
17:26:48.734 MQT: stat/tasmota/STATUS = {"Status":{"Module":75,"DeviceName":"Tasmota","FriendlyName":["Tasmota"],"Topic":"tasmota","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[15,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0}}
17:26:48.742 MQT: stat/tasmota/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/tasmota-zbbridge.bin.gz","RestartReason":"Software/System restart","Uptime":"0T02:44:09","StartupUTC":"2021-12-08T13:42:39","Sleep":50,"CfgHolder":4617,"BootCount":11,"BCResetTime":"2021-11-10T20:21:17","SaveCount":45,"SaveAddress":"FC000"}}
17:26:48.748 MQT: stat/tasmota/STATUS2 = {"StatusFWR":{"Version":"10.0.0.4(zbbridge)","BuildDateTime":"2021-12-08T10:03:39","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":160,"Hardware":"ESP8266EX","CR":"383/699"}}
17:26:48.754 MQT: stat/tasmota/STATUS3 = {"StatusLOG":{"SerialLog":0,"WebLog":2,"MqttLog":3,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["PRIVATE.LOBBYSERVER.LOCAL",""],"TelePeriod":60,"Resolution":"558180C0","SetOption":["0002800D","2805C80001000600003C5A0A000000000000","00000280","00006000","00004001"]}}
17:26:48.764 MQT: stat/tasmota/STATUS4 = {"StatusMEM":{"ProgramSize":671,"Free":1120,"Heap":22,"ProgramFlashSize":2048,"FlashSize":2048,"FlashChipId":"1540A1","FlashFrequency":40,"FlashMode":3,"Features":["00000809","0F1007C6","04400001","00000003","00000000","00000000","00020080","00200000","00000000"],"Drivers":"1,2,4,7,9,10,12,20,23,38,41,50","Sensors":"1"}}
17:26:48.774 MQT: stat/tasmota/STATUS5 = {"StatusNET":{"Hostname":"tasmota-7196","IPAddress":"172.16.1.130","Gateway":"172.16.1.1","Subnetmask":"255.255.255.0","DNSServer1":"172.16.1.11","DNSServer2":"0.0.0.0","Mac":"98:F4:AB:E2:FC:1C","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
17:26:48.781 MQT: stat/tasmota/STATUS6 = {"StatusMQT":{"MqttHost":"172.16.1.13","MqttPort":1883,"MqttClientMask":"tasmota","MqttClient":"tasmota","MqttUser":"tasmota","MqttCount":2,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
17:26:48.789 MQT: stat/tasmota/STATUS7 = {"StatusTIM":{"UTC":"2021-12-08T16:26:48","Local":"2021-12-08T17:26:48","StartDST":"2021-03-28T02:00:00","EndDST":"2021-10-31T03:00:00","Timezone":"+01:00","Sunrise":"08:30","Sunset":"16:53"}}
17:26:48.794 MQT: stat/tasmota/STATUS10 = {"StatusSNS":{"Time":"2021-12-08T17:26:48"}}
17:26:48.802 MQT: stat/tasmota/STATUS11 = {"StatusSTS":{"Time":"2021-12-08T17:26:48","Uptime":"0T02:44:09","UptimeSec":9849,"Vcc":3.458,"Heap":22,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":20,"MqttCount":2,"Wifi":{"AP":1,"SSId":"PRIVATE.LOBBYSERVER.LOCAL","BSSId":"FC:EC:DA:3B:A0:4F","Channel":1,"Mode":"11n","RSSI":100,"Signal":-37,"LinkCount":2,"Downtime":"0T00:00:04"}}}


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._
Show the current power usage, meaby the posibility to toggle the on of state of the device, but i dont know if thats possible at all with the zigbee protocol.

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._
data received when device state is changes from on to off: 
`17:28:28.562 MQT: tele/tasmota/SENSOR = {"ZbReceived":{"0xC958":{"Device":"0xC958","Power":1,"Endpoint":1,"LinkQuality":168}}}`

### ADDITIONAL CONTEXT
_Add any other context about the problem here._
Meaby, if you are from the Netherlands I can send you a power plug by post, so you can test and debug, But i received this plugs with 2 weeks when I ordered it. 

Let me know if you need aditional information or i need to test something for you

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by sfromis** [comment]
Well...... Tuya Zigbee is not compatible with Zigbee. It may be possible to get it to work with some extra steps, not my expertise.

**Reply by WoutervanderLoopNL** [reply]
Whats the different with Tuya Zigbee and Zigbee itself? its just a protocol right?

**Reply by s-hadinger** [reply]
Yes. Zigbee standards defines messages for many devices types. It's called ZCL - Zigbee Cluster Library.

Tuya Zigbee ignores ZCL and is more or less a Zigbee encapsulation of the Tuya Serial protocol - which means it's outside of any standard, and undocumented.

**Reply by WoutervanderLoopNL** [reply]
ah, didnt know that, good to know.  Than i guess its my fault I didnt do my research before I bought 10 of those smart plugs,

i buy a tuya zigbee bridge now, to hopfully make it work with a tuya intergration.

**Comment by WoutervanderLoopNL** [comment]
Any idea when this support comes? ive bought for 100 euros on that power plugs in the understanding that this was siply supported, lol

know I found out it isn't 

Which products are supported to acomplish that i can monitor power of some devices in Homeassistant (via Tasmota)

**Reply by ascillato** [reply]
> Which products are supported to acomplish that i can monitor power of some devices in Homeassistant (via Tasmota)

https://zigbee.blakadder.com/

**Comment by kamiKAC** [comment]
Looks like your device is similar (or the same) as mine. My topic is here: https://github.com/arendst/Tasmota/discussions/13776
When I've started to ask for help I probably had plug without power meter.
Today I got another one (the same type) and I have some success as I was able to read mains voltage. So maybe not everything is lost.
Look at my topic.
